src/wrappers/python/CMakeLists.txt

```
@@ -1,23 +1,37 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright Contributors to the OpenEXR Project.
 
+message(STATUS "Building OpenEXR python bindings")
+
 if(NOT "${CMAKE_PROJECT_NAME}" STREQUAL "OpenEXR")
   cmake_minimum_required(VERSION 3.12)
   project(PyOpenEXR)
   find_package(OpenEXR)
 endif()
 
-add_library (PyOpenEXR SHARED OpenEXR.cpp)
+find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
+
+python_add_library (PyOpenEXR MODULE OpenEXR.cpp)
 
-set (Python_ADDITIONAL_VERSIONS 3)
-find_package (PythonLibs REQUIRED)
-find_package (PythonInterp REQUIRED)
+target_link_libraries (PyOpenEXR PRIVATE "${Python_LIBRARIES}" OpenEXR::OpenEXR)
 
-include_directories ("${PYTHON_INCLUDE_DIRS}")
+# The python module should be called "OpenEXR.so", not "PyOpenEXR.so",
+# but "OpenEXR" is taken as a library name by the main lib, so specify
+# the name explicitly here.
 
-set_target_properties (PyOpenEXR PROPERTIES PREFIX "")
 set_target_properties (PyOpenEXR PROPERTIES OUTPUT_NAME "OpenEXR")
-set_target_properties (PyOpenEXR PROPERTIES SUFFIX ".so")
 
-target_link_libraries (PyOpenEXR "${PYTHON_LIBRARIES}" OpenEXR::OpenEXR)
+configure_file(Imath.py ${CMAKE_CURRENT_BINARY_DIR}/Imath.py COPYONLY)
+
+install(TARGETS PyOpenEXR DESTINATION python/OpenEXR)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Imath.py DESTINATION python/OpenEXR)
```

```suggestion
set(PYTHON_INSTALL_DIR "python/OpenEXR")
if(SKBUILD)
  set(PYTHON_INSTALL_DIR ${SKBUILD_PLATLIB_DIR})
endif()

install(TARGETS PyOpenEXR DESTINATION ${PYTHON_INSTALL_DIR})
install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Imath.py DESTINATION ${PYTHON_INSTALL_DIR})
```

scikit-build-core runs `cmake --install <build path> --prefix <tmpo install path/platlib --strip` by default.

What you put in `platlib` will be stored in the root of the wheel. Wheels are very simple and an installer (pip) simply unpacks the content into `site-prefixes`. (That's an over-simplified explanation). So if we want to be able to `pip install OpenEXR` and run `import OpenEXR`, we need put it at the root (without `python/` prefix), else we will endup with a package importable as `python`.

Note that I'm not sure if your entention was to be able to do `import OpenEXR.OpenEXR` and `import OpenEXR.Imath` or more `import OpenEXR` and `import Imath`.

---

The intention is `import OpenEXR`.  Are you recommending changing the above `python/OpenEXR` to just `OpenEXR`, or does your suggestion support `import OpenEXR` itself?

---

Then. in the case where `SKBUILD` is not defined (so we are not building the wheels), it's up to you to decide if you want to nest `OpenEXR.so` and `Imath.py` into `python/OpenEXR/` or just `python/` or `OpenEXR/`. SO I don't have any specific recommendation for the non wheel build scenario.

---

Got it, thanks. I'll merge your suggestion and test it out myself.